src/Stratis.Bitcoin.Features.BlockStore/BlockRepository.cs

```diff
@@ -216,6 +223,12 @@ public Transaction[] GetTransactionsByIds(uint256[] trxids, CancellationToken ca
                         continue;
                     }
 
+                    if (this.genesisTransactions.TryGetValue(trxids[i], out Transaction genesisTransaction))
```

Comment:
@rowandh was this done like this to act as some sort of cache?

Reply:
Exactly. Saves us deserializing the genesis block + computing the transaction hashes repeatedly.